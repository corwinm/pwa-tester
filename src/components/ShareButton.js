import React from "react";

export default function ShareButton() {
  const share = async () => {
    if (navigator.share) {
      try {
        await navigator.share({
          title: "PWA Tester",
          text: "PWA Tester",
          url: window.location.href
        });
        console.log("Successful share.");
      } catch (error) {
        console.log("Error sharing:", error);
      }
    } else {
      console.log("Share not supported.");
    }
  };

  return (
    <button className="button is-primary" onClick={share} disabled={!navigator.share}>
      <span className="icon">
        <i className="fas fa-share" />
      </span>
      <span>Share</span>
    </button>
  );
}
