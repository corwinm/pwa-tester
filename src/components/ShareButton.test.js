import React from "react";
import { render, cleanup, fireEvent } from "react-testing-library";
import ShareButton from "./ShareButton";

afterEach(cleanup);

test("matches enabled snapshot", () => {
  navigator.share = jest.fn();
  const { container } = render(<ShareButton />);
  expect(container.firstChild).toMatchSnapshot();
});

test("matches disabled snapshot", () => {
  navigator.share = undefined;
  const { container } = render(<ShareButton />);
  expect(container.firstChild).toMatchSnapshot();
});

test("clicking share calls share api",() => {
  navigator.share = jest.fn();
  const { getByText } = render(<ShareButton />);
  fireEvent.click(getByText(/share/i));
  
  expect(navigator.share).toBeCalled();
})