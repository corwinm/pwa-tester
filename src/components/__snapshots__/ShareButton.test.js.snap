// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches disabled snapshot 1`] = `
<button
  class="button is-primary"
  disabled=""
>
  <span
    class="icon"
  >
    <i
      class="fas fa-share"
    />
  </span>
  <span>
    Share
  </span>
</button>
`;

exports[`matches enabled snapshot 1`] = `
<button
  class="button is-primary"
>
  <span
    class="icon"
  >
    <i
      class="fas fa-share"
    />
  </span>
  <span>
    Share
  </span>
</button>
`;
